src/compute/sort/lex_sort.rs

```diff
@@ -135,9 +118,12 @@ pub(crate) fn build_compare(array: &dyn Array, sort_option: SortOptions) -> Resu
     })
 }
 
-/// Sort elements lexicographically from a list of `ArrayRef` into an unsigned integer
-/// [`Int32Array`] of indices.
-pub fn lexsort_to_indices(columns: &[SortColumn]) -> Result<PrimitiveArray<i32>> {
+/// Sorts a list of [`SortColumn`] into a non-nullable [`PrimitiveArray<i32>`]
+/// representing the indices that would sort the columns.
```

Comment:
Why not return u32 instead of i32, [0,65536) is very small range for indices in bigdata area.

Reply:
The official arrow used UInt32Array.

Reply:
I agree. This was due to a limitation that I meanwhile lifted. Follow up PR I will make this sort generic over index, which means that I plan to support `u32, i32, u64, i64` at the users' discretion.

Reply:
Done here: https://github.com/jorgecarleitao/arrow2/pull/220 Thanks for the input :)